libraries/AP_OpenDroneID/AP_OpenDroneID.cpp

```diff
@@ -0,0 +1,546 @@
+        float height = create_altitude((current_location.alt - _operator_position.alt)*0.01f);
+                                                               //height needs to be set depending on the height reference.
+                                                               //If set to MAV_ODID_HEIGHT_REF_OVER_TAKEOFF, height is the difference,
+                                                               //between current altitude and take off height
+
+//set to dummy values as AHRS class does not provide these metrics
+        int16_t speed_vertical = 0;
+        speed_vertical = create_speed_vertical(speed_vertical); //make sure speed is within Remote ID limit
+        uint8_t horizontal_accuracy = MAV_ODID_HOR_ACC_UNKNOWN; //ahrs class does not provide accuracy readings
+        horizontal_accuracy = create_enum_horizontal_accuracy(5.0); //set to 5.0 meter accuracy and use helper function to determine accuracy type.
+        uint8_t vertical_accuracy = MAV_ODID_VER_ACC_UNKNOWN; //ahrs class does not provide accuracy readings
+        vertical_accuracy = create_enum_vertical_accuracy(5.0); //set to 5.0 meter accuracy and use helper function to determine accuracy type.
+        uint8_t barometer_accuracy = MAV_ODID_VER_ACC_UNKNOWN; //ahrs class does not provide accuracy readings
+        barometer_accuracy = create_enum_vertical_accuracy(1.0); //set to 1.0 meter accuracy and use helper function to determine accuracy type.
+        uint8_t speed_accuracy = MAV_ODID_SPEED_ACC_UNKNOWN; //ahrs class does not provide accuracy readings
+        speed_accuracy = create_enum_speed_accuracy(3.0); //set to 3.0 meter accuracy and use helper function to determine accuracy type.
+        uint8_t timestamp_accuracy = MAV_ODID_TIME_ACC_UNKNOWN;
+        timestamp_accuracy =  create_enum_timestamp_accuracy(1.0); //set to 1 sec accuracy
+//end dummy values
+
+        //timestamp is in format seconds (up to one hour).microseconds e.g. 164.47, Maximum value = 3600
+        uint32_t seconds = (AP_HAL::millis()/1000) % 3600;
+        uint32_t milli_seconds = ((AP_HAL::millis()) % 1000);
+        float timestamp = (float) seconds + (float) milli_seconds/1000.0;
+        timestamp = create_location_timestamp(timestamp); //make sure timestamp is within Remote ID limit
+
+        MAV_ODID_STATUS status = get_uav_status();
+        MAV_ODID_HEIGHT_REF height_reference = get_height_reference();
+        const uint8_t* id_or_mac = _id_or_mac;
+        mavlink_channel_t mavlink_channel = get_mavlink_channel();
+
+        mavlink_msg_open_drone_id_location_send(
+            mavlink_channel,
+            0,                  // System ID (0 for broadcast)
+            0,                  // Component ID (0 for broadcast)
+            id_or_mac,                  // id_or_mac: unused. Only used for drone ID data received from other UAs
+            status,
+            direction,
```

Comment:
What is the unit of this? The MAVLink message expects centi-degrees. Based on the limit values, I don't think that is what is currently passed.

Reply:
fixed